File: python/file.py
import dataclasses
import os

@dataclasses.dataclass
class File:
    path: str
    content: str

def read_local_file(path: str, location: str=None) -> File:
    if location:
        print("location", location)
        directory_path = os.path.relpath(os.path.dirname(location))
        path = os.path.join(directory_path, path)
    with open(path) as f:
        return File(path, f.read())

def virtual_file(path: str, content: str) -> File:
    return File(path, content)

File: js/src/file.ts
import { readFileSync } from "fs"
import { join } from "path"

import type { File } from "./structure"

export type { File }

export function readLocalFile(path: string, dirname: string): File {
    let content = readFileSync(join(dirname, path), "utf-8")
    return { path, content }
}
